### PR TITLE
[AKS]: Remove 200/OK response from reset SP/AAD api

### PR DESCRIPTION
**Pull request by zqingqing1**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
Need confirmation from @ravbhatnagar that returning 202 from a synchronous operation is ok.

File: specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json [hunk 1]
```diff
@@ -581,7 +575,6 @@
             }
           }
         },
-        "x-ms-long-running-operation": true,
```
[the remark this answers is not in the document]
Comment (zqingqing1) [reply specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json]: It is a long-running operation, but we only return back the 202 response to customer.

[the remark this answers is not in the document]
Comment (jhendrixMSFT) [reply specification/containerservices/resource-manager/Microsoft.ContainerService/preview/2018-08-01-preview/managedClusters.json]: There is no `Location` or `Azure-AsyncOperation` header being returned?

**Comment by jhendrixMSFT** [comment]
@zqingqing1 can you please explain why this is being removed?  It's introducing new SDK validation failures, see [here](https://github.com/Azure/azure-rest-api-specs/pull/4948/checks?check_run_id=43074728).  Current requirement is to model the terminal LRO state in the swagger.

**Comment by zqingqing1** [comment]
The reason why it is been removed, because 200 is not in the response status of these 2 APIs. So right now, I am thinking maybe we can return operation status as 200 after polling header, if we wanna resolve this error.

**Comment by jhendrixMSFT** [comment]
@lmazuel I was told you asked to have this validation rule put in place.  Can you please explain the details behind it?

**Comment by zqingqing1** [comment]
@jhendrixMSFT, I added 200 response to resolve the error. Please review :)

**Comment by jhendrixMSFT** [comment]
@zqingqing1 there is a merge conflict, can you please resolve it?

**Comment by jhendrixMSFT** [comment]
With these changes the generated code no longer returns a `ManagedCluster` object when the LRO completes.  Take a look at the [Go codegen](https://github.com/Azure/azure-sdk-for-go/pull/3829/files#diff-3cc4ae0a07e198ede58c0be8ee523998) for an example.  Is this expected?

**Comment by zqingqing1** [comment]
Yes, it is expected.

**Comment by jhendrixMSFT** [comment]
So when the polling completes the user needs to make a separate API call then to retrieve the result?

**Comment by zqingqing1** [comment]
Yes, that is right.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**